Update watchdog explains (#30782)
DataDog/documentation: content/en/dashboards/graph_insights/watchdog_explains.md

To account for seasonality, the algorithm looks back up to three weeks in time. For example, if a spike appears on a Monday at 9:00 a.m., Watchdog compares that datapoint against previous Mondays at the same hour. If similar patterns appear consistently, the spike is treated as **seasonal** and not flagged as an anomaly. This helps reduce false positives and ensures that only unexpected deviations are surfaced.

Anomalies can be sharp spikes or drops, but may also be more subtle trends like step changes or slope shifts.

<div class="alert alert-info">Anomaly detection in Watchdog Explains works with <strong>Metrics data</strong> (avg, sum, min, and max aggregation).</div>

## Investigating anomalies

{{< img src="/dashboards/graph_insights/watchdog_explains/investigate_anomaly.png" alt="Watchdog Explains investigation view showing anomaly detection on a dashboard" style="width:100%;" >}}

You can start your investigation from any timeseries graph that uses metric data. When Watchdog Explains detects an anomaly, it highlights the affected region with a pink box. To begin investigating, click **Investigate Anomaly**.

> This opens a full-screen investigation view where Watchdog analyzes the anomaly and surfaces any tag groups that significantly contributed to the shape or scale of the anomaly. Click on a tag to see how removing or isolating that dimension affects the graph. Use this to identify root causes like specific customers, services, or environments.

Reviewer: ```suggestion
This opens a full-screen investigation view. Watchdog analyzes the anomaly and surfaces any tag groups that significantly contributed to the shape or scale of the anomaly. Click on a tag to see how removing or isolating that dimension affects the graph. Use this to identify root causes like specific customers, services, or environments.
```

## Isolating the cause with tag breakdown

{{< img src="/dashboards/graph_insights/watchdog_explains/tag_breakdown.png" alt="Tag breakdown visualization in Watchdog Explains" style="width:100%;" >}}

Watchdog Explains analyzes the tag dimensions of the query to determine what may have caused the shift. It re-runs the same graph query filtered by each tag group (such as service, availability zone, or customer tier) to isolate the segments most likely responsible.

By comparing the shape of the filtered timeseries with the overall graph, Watchdog Explains can infer which tag group had the greatest impact.

> When a tag is flagged, it means filtering by that tag significantly reduced or reshaped the anomaly. Contribution is determined by comparing the shape and scale of filtered timeseries against the original graph. Not every anomaly has an obvious root cause. If no tag group meaningfully contributes to the spike or dip, Watchdog Explains may surface a general anomaly without tag-level attribution.

Reviewer: ```suggestion
When a tag is flagged, it means filtering by that tag significantly reduced or reshaped the anomaly. Watchdog compares the filtered timeseries to the original graph to pinpoint which tags contributed to the anomaly most. Sometimes, an anomaly may not have an obvious root cause. If no tag group meaningfully contributes to the spike or dip, Watchdog Explains may surface a general anomaly without tag-level attribution.
```
